airflow/bin/cli.py

```diff
@@ -833,136 +833,159 @@ def start_refresh(gunicorn_master_proc):
+                        signal.SIGINT: kill_proc,
+                        signal.SIGTERM: kill_proc
+                    },
+                )
+                with ctx:
+                    subprocess.Popen(run_args, close_fds=True)
+
+                    # Reading pid file directly, since Popen#pid doesn't
+                    # seem to return the right value with DaemonContext.
+                    while True:
+                        try:
+                            with open(pid) as f:
+                                gunicorn_master_proc_pid = int(f.read())
+                                break
+                        except IOError:
+                            log.debug("Waiting for gunicorn's pid file to be created.")
+                            time.sleep(0.1)
+
+                    gunicorn_master_proc = psutil.Process(gunicorn_master_proc_pid)
+                    self.monitor_gunicorn(gunicorn_master_proc)
+
+                stdout.close()
+                stderr.close()
+            else:
+                gunicorn_master_proc = subprocess.Popen(run_args, close_fds=True)
 
+                signal.signal(signal.SIGINT, kill_proc)
+                signal.signal(signal.SIGTERM, kill_proc)
+
+                self.monitor_gunicorn(gunicorn_master_proc)
+
+    def prepare_run_args(self, pid):
+        """ combine command to run gunicorn server """
         run_args = [
             'gunicorn',
-            '-w', str(num_workers),
-            '-k', str(args.workerclass),
-            '-t', str(worker_timeout),
-            '-b', args.hostname + ':' + str(args.port),
+            '-w', str(self.args.workers),
+            '-k', str(self.args.workerclass),
+            '-t', str(self.args.worker_timeout),
+            '-b', self.args.hostname + ':' + str(self.args.port),
             '-n', 'airflow-webserver',
             '-p', str(pid),
             '-c', 'python:airflow.www.gunicorn_config',
         ]
 
-        if args.access_logfile:
-            run_args += ['--access-logfile', str(args.access_logfile)]
+        if self.args.access_logfile:
+            run_args += ['--access-logfile', str(self.args.access_logfile)]
 
-        if args.error_logfile:
-            run_args += ['--error-logfile', str(args.error_logfile)]
+        if self.args.error_logfile:
+            run_args += ['--error-logfile', str(self.args.error_logfile)]
 
-        if args.daemon:
+        if self.args.daemon:
             run_args += ['-D']
 
-        if ssl_cert:
-            run_args += ['--certfile', ssl_cert, '--keyfile', ssl_key]
+        if self.args.ssl_cert:
+            run_args += ['--certfile', self.args.ssl_cert, '--keyfile',
+                         self.args.ssl_key]
+
+        if self.args.gunicorn_config:
+            if isinstance(self.args.gunicorn_config, list):
+                self.args.gunicorn_config = self.args.gunicorn_config[0]
+            for arg in self.args.gunicorn_config.split():
+                run_args.append(arg)
```

Comment:
Is this used as:

```
airflow webserver '--gunicorn-opt-a --gunicorn-opt-b'
```

If so why not just take the list directly, make it multiple args not a single one, remove the need for lines 955-958 

```suggestion
            run_args += self.args.gunicorn_args
```